rbac_api.go

```diff
@@ -171,14 +171,25 @@ func (e *Enforcer) GetImplicitRolesForUser(name string, domain ...string) []stri
 //
 // GetPermissionsForUser("alice") can only get: [["alice", "data2", "read"]].
 // But GetImplicitPermissionsForUser("alice") will get: [["admin", "data1", "read"], ["alice", "data2", "read"]].
-func (e *Enforcer) GetImplicitPermissionsForUser(user string) [][]string {
-	roles := e.GetImplicitRolesForUser(user)
-	roles = append([]string{user}, roles...)
-
+func (e *Enforcer) GetImplicitPermissionsForUser(user string, domain ...string) [][]string {
+	roles := e.GetImplicitRolesForUser(user, domain...)
 	res := [][]string{}
+	permissions := [][]string{}
+
 	for _, role := range roles {
-		permissions := e.GetPermissionsForUser(role)
+		permissions = e.GetPermissionsForUser(role)
 		res = append(res, permissions...)
 	}
```

Comment:
Why there's no domain here? I mean why don't use ``GetPermissionsForUserInDomain()``?

Reply:
Because call GetImplicitRolesForUser with domain parameter will return roles of user, so call GetPermissionsForUser is without domain parameter.